src/charm.py

```diff
@@ -105,6 +107,16 @@ def _on_leader_elected(self, event) -> None:
 
         self._update_hosts(event)
 
+        if "db_initialised" not in self.app_data:
```

Comment:
since you are testing this on `update_app_relation_data` and not testing on `_relation_departed`, you could remove it from here for code consistency

Reply:
makes sense. thanks Paulo!

Reply:
Actually I think it should be kept here, when the charm starts up for the first time it executes the `on_leader_elected` hook before the start hook and so the DB won't have been started i.e.  "db_initialised" won't be in self.app_data and doing something like `update_app_relation_data` could cause problems since the functions in `update_app_relation_data` use communication with `mongod`

Reply:
@MiaAltieri is absolutely right

Reply:
Just discussed with Mia, the functionality mentioned is already there. Will simply remove redundant check in leader election handler.

Reply:
ack